src/main/java/com/vaadin/guice/server/VaadinModule.java

```diff
@@ -39,6 +40,7 @@ public VaadinModule(SessionProvider sessionProvider, Set<Class<? extends View>>
     @Override
     protected void configure() {
         bindScope(UIScope.class, uiScoper);
+        bindScope(GuiceUI.class, uiScoper);
```

Comment:
the documentation was copied from vaadin-spring, but we should stay as close to spring as possible so I guess this is the right way to go. Although I don't really see a point in putting the UI itself in the UI-Scope.